src/main/java/seedu/address/model/person/Position.java

```diff
@@ -0,0 +1,72 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents an Organisation's position (that the user is applying to) in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidPosition(String)}
+ */
+public class Position {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Positions should only contain alphanumeric characters and spaces, and it should not be blank";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ][^(?![\\s\\S])]*";
```

Comment:
The trailing `[^(?![\\s\\S])]*` doesn't appear to do anything useful other than making it incorrectly more relaxed, so something like `Abc123 +123` is a valid input despite what `MESSAGE_CONSTRAINTS` suggest.

```suggestion
    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
```